apps/api/v2/src/modules/workflows/workflows.repository.ts

```diff
@@ -0,0 +1,305 @@
+      },
+      input: await this.mapCreateDtoToZodUpdateSchema(data, workflowId, teamId, currentWorkflow),
+    });
+
+    const workflow = await this.getTeamWorkflowById(teamId, workflowId);
+    return workflow;
+  }
+
+  public toDto(workflow: WorkflowType): WorkflowOutput {
+    const activation: WorkflowActivationDto = {
+      isActiveOnAll: workflow.isActiveOnAll,
+
+      activeOnEventTypeIds: workflow.activeOn?.map((relation) => relation.eventTypeId) ?? [],
+    };
+
+    const trigger: WorkflowTriggerDto = {
+      type: workflow.trigger as WorkflowTriggerType,
+      offset:
+        workflow.time !== null && workflow.timeUnit !== null
+          ? { value: workflow.time, unit: workflow.timeUnit as WorkflowTimeUnit }
+          : undefined,
+    };
+
+    const steps: WorkflowStepDto[] = workflow.steps.map((step) => {
+      const message: WorkflowMessageDto = {
+        subject: step.emailSubject ?? "",
+
+        text: "",
+        html: "",
+      };
+
+      let recipient: RecipientType;
+      let email = "";
+      let phone = "";
+
+      switch (step.action as StepAction) {
+        case StepAction.EMAIL_HOST:
+          recipient = RecipientType.HOST;
+          message.html = step.reminderBody ?? "";
+          break;
+        case StepAction.EMAIL_ATTENDEE:
+          message.html = step.reminderBody ?? "";
+          recipient = RecipientType.ATTENDEE;
+          break;
+        case StepAction.SMS_ATTENDEE:
+          message.text = step.reminderBody ?? "";
+          recipient = RecipientType.ATTENDEE;
+          break;
+        case StepAction.WHATSAPP_ATTENDEE:
+          message.text = step.reminderBody ?? "";
+
+          recipient = RecipientType.ATTENDEE;
+          break;
+        case StepAction.EMAIL_ADDRESS:
+          message.html = step.reminderBody ?? "";
+          recipient = RecipientType.EMAIL;
+          email = step.sendTo ?? "";
+          break;
+        case StepAction.SMS_NUMBER:
+        case StepAction.WHATSAPP_NUMBER:
+          message.text = step.reminderBody ?? "";
+          recipient =
+            step.action === StepAction.SMS_NUMBER ? RecipientType.PHONE_NUMBER : RecipientType.PHONE_NUMBER;
+          phone = step.sendTo ?? "";
+          break;
+        default:
+          recipient = RecipientType.ATTENDEE;
+      }
+
+      return {
+        id: step.id,
+        stepNumber: step.stepNumber,
+        action: step.action as StepAction,
+        recipient: recipient,
+        email,
+        phone,
+        template: step.template as TemplateType,
+        includeCalendarEvent: step.includeCalendarEvent,
+        sender: step.sender ?? "Default Sender",
+        message: message,
+      };
+    });
+
+    return {
+      id: workflow.id,
+      name: workflow.name,
+      activation: activation,
+      trigger: trigger,
+      steps: steps,
+    };
+  }
+
+  public async mapCreateDtoToZodUpdateSchema(
+    createDto: Partial<CreateWorkflowDto>,
```

Comment:
If we have clear input DTOs like i wrote above we would not need to pass a partial and then check it here manually.